src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/SnikeIKøenService.kt

```diff
@@ -0,0 +1,175 @@
+package no.nav.familie.ks.sak.kjerne.behandling
+
+import no.nav.familie.ks.sak.kjerne.behandling.domene.Behandling
+import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingStatus
+import no.nav.familie.ks.sak.kjerne.behandling.steg.BehandlingStegStatus
+import no.nav.familie.ks.sak.kjerne.logg.LoggService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.Clock
+import java.time.LocalDateTime
+
+@Service
+class SnikeIKøenService(
+    private val behandlingService: BehandlingService,
+    private val loggService: LoggService,
+    private val tilbakestillBehandlingService: TilbakestillBehandlingService,
+    private val clock: Clock
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Transactional
+    fun settAktivBehandlingPåMaskinellVent(
+        behandlingId: Long,
+        årsak: SettPåMaskinellVentÅrsak,
+    ) {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+
+        if (!behandling.aktiv) {
+            throw IllegalStateException("Behandling=$behandlingId er ikke aktiv")
+        }
+
+        val behandlingStatus = behandling.status
+        val behandlingStegTilstand = finnBehandlingStegTilstand(behandling)
+
+        val erBehandlingIkkePåVent = behandlingStegTilstand.behandlingStegStatus !== BehandlingStegStatus.VENTER
+
+        if (erBehandlingIkkePåVent && behandlingStatus !== BehandlingStatus.UTREDES) {
+            throw IllegalStateException("Behandling=$behandlingId kan ikke settes på maskinell vent då status=$behandlingStatus")
+        }
+
+        behandling.status = BehandlingStatus.SATT_PÅ_MASKINELL_VENT
+        behandling.aktiv = false
+        behandlingService.oppdaterBehandling(behandling)
+        loggService.opprettSettPåMaskinellVent(behandling, årsak.beskrivelse)
+    }
+
+    private fun finnBehandlingStegTilstand(behandling: Behandling) =
+        behandling.behandlingStegTilstand.single { it.behandlingSteg == behandling.steg }
```

Comment:
Blir denne logikken helt riktig? Er det slik at det siste innslaget i "behandlingStegTilstand" settet er det som er gjeldende uavhengig av hvilket steg man befinner seg i? Jeg ser fp-sak har bla. en metode for å kun hente "siste" `BehandlingStegTilstand` https://github.com/navikt/fp-sak/blob/master/behandlingslager/domene/src/main/java/no/nav/foreldrepenger/behandlingslager/behandling/Behandling.java#L388 

Hadde det vært mer riktig å bruke en slik metode?